app/assets/javascripts/discourse/app/components/edit-category-general.gjs

```diff
@@ -0,0 +1,347 @@
+import Component from "@glimmer/component";
+import { cached } from "@glimmer/tracking";
+import { fn, hash } from "@ember/helper";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { htmlSafe } from "@ember/template";
+import { eq } from "truth-helpers";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import categoryBadge from "discourse/helpers/category-badge";
+import { categoryBadgeHTML } from "discourse/helpers/category-link";
+import { CATEGORY_STYLE_TYPES } from "discourse/lib/constants";
+import getURL from "discourse/lib/get-url";
+import Category from "discourse/models/category";
+import { i18n } from "discourse-i18n";
+import ColorInput from "admin/components/color-input";
+import CategoryChooser from "select-kit/components/category-chooser";
+import ColorPicker from "./color-picker";
+
+export default class EditCategoryGeneral extends Component {
+  @service router;
+  @service site;
+  @service siteSettings;
+
+  uncategorizedSiteSettingLink = getURL(
+    "/admin/site_settings/category/all_results?filter=allow_uncategorized_topics"
+  );
+  customizeTextContentLink = getURL(
+    "/admin/customize/site_texts?q=uncategorized"
+  );
+  foregroundColors = ["FFFFFF", "000000"];
+
+  get styleTypes() {
+    return Object.keys(CATEGORY_STYLE_TYPES).map((key) => ({
+      id: key,
+      name: i18n(`category.styles.options.${key}`),
+    }));
+  }
+
+  get showWarning() {
+    return this.args.category.isUncategorizedCategory;
+  }
+
+  @cached
+  get backgroundColors() {
+    const categories = this.site.get("categoriesList");
+    return this.siteSettings.category_colors
+      .split("|")
+      .map((i) => i.toUpperCase())
+      .concat(categories.map((c) => c.color.toUpperCase()))
+      .uniq();
+  }
+
+  @cached
+  get usedBackgroundColors() {
+    const categories = this.site.get("categoriesList");
```

Comment:
how all of this works with async category loading setting? (when you have lots of categories basically)